lib/image_optim/cmd.rb

```diff
@@ -44,6 +80,42 @@ def check_status!
 
         fail SignalException, status.termsig
       end
+
+      def init_options!(args)
+        pgroup_opt = Gem.win_platform? ? :new_pgroup : :pgroup
+
+        if args.last.is_a?(Hash)
+          args.last[pgroup_opt] = true
+        else
+          args.push(pgroup_opt => true)
+        end
+      end
+
+      def cleanup_process(pid)
+        Thread.new do
+          begin
+            Process.kill('-TERM', pid)
+            Process.detach(pid)
+            now = Time.now
+
+            while Time.now - now < 10
+              begin
+                Process.kill(0, pid)
+                sleep 0.001
```

Comment:
You can extract it as a constant, but I don't currently see a need to make it configurable. Though I would make sleep time bigger, 0.1 or 0.01, may be better to also put it in a constant.
I've wrote code to check the behaviour: [gist](https://gist.github.com/toy/8591fe65c5550755a86dc724a85c0939)